embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/network/logging/EmbraceNetworkLoggingService.kt

```diff
@@ -166,88 +168,66 @@
-        }
-
-        logger.logDeveloper("EmbraceNetworkLoggingService", "Domain: $domain")
+        } ?: return
 
         if (isIpAddress(domain)) {
-            logger.logDeveloper("EmbraceNetworkLoggingService", "Domain is an ip address")
-            val captureLimit = configService.networkBehavior.getNetworkCaptureLimit()
-
-            if (ipAddressCount.getAndIncrement() < captureLimit) {
-                // only capture if the ipAddressCount has not exceeded defaultLimit
-                logger.logDeveloper("EmbraceNetworkLoggingService", "capturing network call")
+            if (ipAddressNetworkCallCount.getAndIncrement() < defaultPerDomainCallLimit) {
                 storeNetworkCall(callId, networkCall)
-            } else {
-                logger.logDeveloper("EmbraceNetworkLoggingService", "capture limit exceeded")
             }
             return
+        } else if (!domainSetting.containsKey(domain)) {
+            createLimitForDomain(domain)
         }
 
-        val settings = domainSettings[domain]
+        val settings = domainSetting[domain]
         if (settings == null) {
-            logger.logDeveloper("EmbraceNetworkLoggingService", "no domain settings")
-            storeNetworkCall(callId, networkCall)
+            // Not sure how this is possible, but in case it is, limit logged logs where we can't figure out the settings to apply
+            if (untrackedNetworkCallCount.getAndIncrement() < defaultPerDomainCallLimit) {
+                storeNetworkCall(callId, networkCall)
+            }
         } else {
             val suffix = settings.suffix
             val limit = settings.limit
-            var count = callsPerDomain[suffix]
+            var countPerSuffix = callsPerDomain[suffix]
 
-            if (count == null) {
-                count = DomainCount(1, limit)
+            if (countPerSuffix == null) {
+                countPerSuffix = DomainCount(0, limit)
             }
 
             // Exclude if the network call exceeds the limit
-            if (count.requestCount < limit) {
+            if (countPerSuffix.requestCount < limit) {
                 storeNetworkCall(callId, networkCall)
             } else {
                 logger.logDeveloper("EmbraceNetworkLoggingService", "capture limit exceeded")
             }
 
             // Track the number of calls for each domain (or configured suffix)
             suffix?.let {
-                callsPerDomain[it] = DomainCount(count.requestCount + 1, limit)
+                callsPerDomain[it] = DomainCount(countPerSuffix.requestCount + 1, limit)
                 logger.logDeveloper(
                     "EmbraceNetworkLoggingService",
-                    "Call per domain $domain ${count.requestCount + 1}"
+                    "Call per domain $domain ${countPerSuffix.requestCount + 1}"
                 )
             }
         }
     }
 
-    private fun storeSettings(url: String) {
+    private fun createLimitForDomain(domain: String) {
```

Comment:
The old implementation is really wasteful in that it rebuild the domain limits map at every call, often unnecessarily. So we build and cache it for reuse at init time and redo it when the session ends instead.